[WD-15094] fix: onhover not working on tablet/mobile meganav

muhammad-ali-pk opened the pull request:
## Done

- Fixed onhover

## QA

- Open this [demo](https://ubuntu-com-14686.demos.haus/) on tablet or mobile screen size
- Open the meganav by clicking the 'menu'
- Navigation to the bottom layer of any navigation item. ex. 'Products' > 'Featured'.
- Hover your mouse over the dropdown items
- Verify hover effect is applied

## Issue / Card

Fixes #[WD-15094](https://warthogs.atlassian.net/browse/WD-15094)


[WD-15094]: https://warthogs.atlassian.net/browse/WD-15094?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

webteam-app commented:
### [<img src='https://assets.ubuntu.com/v1/6baef514-ubuntu-circle-of-friends-large.svg' height=32 width=32> Demo</img>](https://ubuntu-com-14686.demos.haus)
### [<img src='https://assets.ubuntu.com/v1/e512b0e2-jenkins.svg' height=32 width=32> Jenkins </img>](https://jenkins.canonical.com/webteam/job/start-demo/12481)
### [<img src='https://assets.ubuntu.com/v1/7144ec6d-logo-jaas-icon.svg' height=32 width=32> demos.haus </img>](https://demos.haus)

petesfrench commented:
This breaks the layout and on-hover effect on desktop:
![image](https://github.com/user-attachments/assets/d0823a58-c1a9-47e3-a5bc-acd216e094d5)

muhammad-ali-pk commented:
@petesfrench Thanks! Fixed.
PS., if you're wondering why I added hover on <li> rather than anchor tag, it's because anchor tag doesn't occupy the full width of <li> so the hover looks weird.

petesfrench commented:
@muhammad-ali-pk No worries. This is the 'old' meganav. We have since migrated to using a [Vanilla version of it](https://vanillaframework.io/docs/patterns/navigation#full-width-dropdowns) on c.com. Hence the need for some custom CSS. Ideally we will migrate the whole nav, but that will require a fair bit of work
